File: mindspace-ai-engine/src/db/redis.py
from redis.asyncio import Redis

from src.config import env
from src.utils.logger import get_logger

logger = get_logger(__name__)


redis: Redis = Redis.from_url(
    env.REDIS_URL,
    decode_responses=True,
    health_check_interval=30,
)


async def ping_redis() -> None:
    await redis.ping()
    logger.info("Successfully connected to Redis")


async def dispose_redis() -> None:
    await redis.aclose()

File: mindspace-be/src/db/redis.ts
import Redis from "ioredis";
import { env } from "../config/env";

const redis = new Redis(env.REDIS_URL, {
    lazyConnect: false,
    maxRetriesPerRequest: 3,
    enableReadyCheck: true,
});

redis.on("error", (e: unknown) => {
    console.error("Redis error:", e);
});

export { redis };
export default redis;

redis.ping()
    .then(() => {
        console.log("Successfully connected to Redis");
    })
    .catch((e: unknown) => {
        console.error("Failed to connect to Redis:", e);
        process.exit(1);
    });
